llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -1731,6 +1735,16 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     for (auto VT : { MVT::v1i1, MVT::v2i1, MVT::v4i1, MVT::v8i1 })
       setOperationAction(ISD::EXTRACT_SUBVECTOR, VT, Custom);
   }
+  if (Subtarget.hasDQI() && Subtarget.hasVLX()) {
+    setOperationAction(ISD::LRINT, MVT::v4f32, Legal);
```

Comment:
Maybe worth using the VT iteration pattern:
```
for (MVT VT : { MVT::v4f32, MVT::v8f32, MVT::v2f64, MVT::v4f64 }) {
  setOperationAction(ISD::LRINT, VT, Legal);
  setOperationAction(ISD::LLRINT, VT, Legal);
}
```

Reply:
Done.